util/docker_build.sh

```diff
@@ -45,5 +45,5 @@ fi
 dir=$(pwd -W 2>/dev/null) || dir=$PWD  # Use Windows path if on Windows
 
 # Run container and build firmware
-docker run --rm $usb_args -v "$dir":/qmk_firmware qmkfm/qmk_firmware \
+docker run --rm $usb_args --interactive --tty -v "/dev:/dev" -v "$dir":/qmk_firmware qmkfm/qmk_firmware \
```

Comment:
```suggestion
docker run --rm -it $usb_args -v "$dir":/qmk_firmware qmkfm/qmk_firmware \
```
This is most commonly done as `docker run --rm -it`, so I think that order should be used. +1 for this change, though.

As for the `/dev` mount, this is already done on line 38 above as part of `usb_args`.

https://github.com/qmk/qmk_firmware/blob/41a050cba592b9d8348a8a2bdf4ec371d2280a07/util/docker_build.sh#L38

Does this not work for you?

Reply:
Yeah, but I figured that since it is a script it is better to be verbose and improve readability.

Regarding usb_args they never came into effect (and I frankly missed them somehow). I'm running native Docker on Gentoo (no docker-machine).

A fix would be better focused towards line 37?

Reply:
I think that anyone who's used Docker for longer than 5 mins will about `-it`. The same argument could otherwise be applied to `-v`, so I think that `-it` is totally fine.

Replace `-v /dev/bus/usb:/dev/bus/usb` on line 38 with `-v /dev:/dev` and let me know if that works for you.

Reply:
`uname` prints `Linux`. I think you're on to something with the volume mounting. I'll make sure to try it out when I get the chance (I don't have my keyboard with me at the moment).

Reply:
Sounds good!

I suspect the root of the problem is udev/eudev on Gentoo. It might put the USB device files somewhere inside `/dev` other than `/dev/bus/usb`. Although I'm not a fan of mounting the entire `/dev` tree in the container, it is the easiest solution. Anyway, try it out when you get the chance and let me know if it works.

Reply:
I think it may be because my (and probably other) keyboards are using serial interfaces as opposed to USB devices.

```
Detecting USB port, reset your controller now...........
Detected controller on USB port at /dev/ttyACM0
```

So the solution is probably as you say to mount `/dev`.

Reply:
Is it working for you now that you've mounted `/dev:/dev`?

Reply:
@vomindoraan Yep! Thanks for the suggestion.

Reply:
Do you want me to revert the `-it` change in favor of #5088 ?

Reply:
Awesome. I'm inclined to say that explicitly mounting just `/dev/bus/usb:/dev/bus/usb` and `/dev/ttyACM*:/dev/ttyACM*` might be a cleaner/safer solution, but I don't think wildcards work in bind mounts, therefore it would require some additional logic in the shell script. So yeah, let's just go with `/dev:/dev`.
Related: https://askubuntu.com/questions/435861/what-is-the-difference-between-dev-tty-and-dev-bus-usb-001-002/984610

The diffs are compatible so there shouldn't be any conflicts. So no need to revert it :)

Reply:
Yep, I think you are right. If wildcards worked I figure they would only mount existing device nodes and not mount new nodes.